### PR TITLE
feat(prometheus): move HA mixin

### PR DESCRIPTION
**Pull request by Duologic**
This should move too.

I'm sneaking in a tiny compat layer for ksonnet-lib, it should just provide compat inside this library, not on all
libraries. And also a fix for using upstream images.

### CODE REVIEW

File: prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet
Comment (dannykopping) [inline prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet:1]: I wonder if it might not make sense to `git mv prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet prometheus/ha-mixin.libsonnet`, thereby maintaining all the history of quite a fundamental file - and then creating a new file in `prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet` with the current contents.

**Review by dannykopping — APPROVED** [review]
One minor suggestion, otherwise LGTM.
You should probably get a review from one of the others with more context than me before merging

File: prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet
Comment (Duologic) [reply prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet]: I think this gets lost in the squash merge anyway. :neutral_face:

Comment (dannykopping) [reply prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet]: Don't think I follow.
If you `git mv`, the path of the file may change but its history will be maintained. Squashing will combine all your changes in this PR only.

File: prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet
Comment (malcolmholmes) [inline prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet:1]: suggestion:

```
(import 'prometheus/ha-mixin.libsonnet')
```

There are a few people using this mixin, by name, this will prevent that from failing.

**Review by malcolmholmes — COMMENTED** [review]
How about adding `withHA():: (import 'ha-mixin.libsonnet'),` to the end of prometheus.libsonnet, and making that the recommended way of loading the feature, rather than invoking it by filename?

File: prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet
Comment (Duologic) [reply prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet:1]: Already there, please check again.

**Comment by Duologic** [comment]
> How about adding `withHA():: (import 'ha-mixin.libsonnet'),` to the end of prometheus.libsonnet, and making that the recommended way of loading the feature, rather than invoking it by filename?

Already there, please check again.

**Review by malcolmholmes — APPROVED** [review]
Given that everything I proposed is actually already there, I think I can approve.

**Pull request merged**
